pkg/types/config/cluster.go

```diff
@@ -0,0 +1,283 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/coreos/tectonic-config/config/tectonic-network"
+	"gopkg.in/yaml.v2"
+
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/config/aws"
+	"github.com/openshift/installer/pkg/types/config/libvirt"
+)
+
+const (
+	// IgnitionPathMaster is the relative path to the ign master cfg from the tf working directory
+	// This is a format string so that the index can be populated later
+	IgnitionPathMaster = "master-%d.ign"
+	// IgnitionPathWorker is the relative path to the ign worker cfg from the tf working directory
+	IgnitionPathWorker = "worker.ign"
+	// PlatformAWS is the platform for a cluster launched on AWS.
+	PlatformAWS Platform = "aws"
+	// PlatformLibvirt is the platform for a cluster launched on libvirt.
+	PlatformLibvirt Platform = "libvirt"
+	// DefaultChannel is the default RHCOS channel for the cluster.
+	DefaultChannel = "tested"
```

Comment:
These are not types, so I'd like to keep them out of `pkg/types/...`.  `DefaultChannel` should probably live in `pkg/rhcos`, the ignition paths should probably live in `pkg/asset/ignition`, etc.

And I think this shift is a really good time to drop the old YAML-specific strings and just write the Terraform with the same JSON properties that we use for YAML, although that will mean updating the consuming Terraform code (e.g. with `tectonic_cluster_cidr` -> `podCIDR`.

Reply:
@wking I'd like to do that in another separate PR, wdyt?

Reply:
> I'd like to do that in another separate PR, wdyt?

Fine with me.